### DISCUSSION TITLE
Powershell Core support

### DISCUSSION THREAD

**Discussion by adrianriobo** [crc-org/crc#2353] [Q&A] [answered]
Can we extend [support](https://github.com/code-ready/crc/blob/808919f7a1d3732ea798641f64d9ad4cbde3db12/pkg/os/shell/shell_windows.go#L12) for Powershell to include [Powershell Core](https://docs.microsoft.com/es-es/powershell/scripting/overview?view=powershell-7.1). 

This is part of the open source adoption from Microsoft, and is quiet reasonable to have this on enterprise companies.

I gave a try to execute crc from this shell and everything work exactly the same (main deviation is the set of cmdlets where Powershell Core try to keep them short, providing capacity to extend them loading specific modules, in our case we do not required nothing of it, so crc works out of the box here).

The question is around [windows shell identification](https://github.com/code-ready/crc/blob/808919f7a1d3732ea798641f64d9ad4cbde3db12/pkg/os/shell/shell_windows.go#L75), the background process where crc is being executed: core powershell process uses `pwsh` instead of `powershell`

[here](https://crcqe-jenkins-csb-crc.apps.ocp4.prod.psi.redhat.com/blue/organizations/jenkins/qe-bundle_baremetal_windows10-brno/detail/qe-bundle_baremetal_windows10-brno/11/pipeline/) results from e2e run on downstream environment using latest Powershell Core on Windows instance

**Comment by gbraad** [comment]
The question is; "does it not work"? We invoke a separate powershell
instance internally. So this should work, right?
The reason we don't do this as the Hyper-V cmdlets needs to be enabled for
use in PSCore, and this is not a change we wan't to make permanently.

On Wed, May 19, 2021 at 12:26 AM Adrián Riobo Lorenzo <
***@***.***> wrote:

> Can we extend support
> <https://github.com/code-ready/crc/blob/808919f7a1d3732ea798641f64d9ad4cbde3db12/pkg/os/shell/shell_windows.go#L12>
> for Powershell to include Powershell Core
> <https://docs.microsoft.com/es-es/powershell/scripting/overview?view=powershell-7.1>
> .
>
> This is part of the open source adoption from Microsoft, and is quiet
> reasonable to have this on enterprise companies.
>
> I gave a try to execute crc from this shell and everything work exactly
> the same (main deviation is the set of cmdlets where Powershell Core try to
> keep them short, providing capacity to extend them loading specific
> modules, in our case we do not required nothing of it, so crc works out of
> the box here).
>
> The question is around windows shell identification
> <https://github.com/code-ready/crc/blob/808919f7a1d3732ea798641f64d9ad4cbde3db12/pkg/os/shell/shell_windows.go#L75>,
> the background process where crc is being executed: core powershell process
> uses pwsh instead of powershell
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/code-ready/crc/discussions/2353>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAAOZT6C2I6GYYNFAUGTSTTOKIKFANCNFSM45C3SVEQ>
> .
>


-- 

   Gerard Braad | http://gbraad.nl

   STEM is the source, but BUILD is how you develop!
   [ Better Understanding Involves Learning and Doing ]

**Reply by adrianriobo** [reply]
As far as I have seen the not working part is the `crc oc-env` as the shell is not detected as powershell due to the identification mechanism, so the command offered with `crc oc-env` is default which is linux bash command. 

As an example if I force `SHELL=powershell` everything work as expected, because this takes precedence on shell identification

**Reply by gbraad** [reply]
In that case an additional shell to include (`pwsh`) would be simple.

**Reply by gbraad** [reply]
@adrianriobo create an issue for this addition, or document this?

**Comment by gbraad** [comment, marked as answer]
PR for dectection has been proposed #3500
